src/serde/tools.rs

```diff
@@ -282,124 +340,116 @@ mod tests {
         );
     }
 
-    #[cfg(test)]
-    mod test {
-        use super::*;
-        use crate::serde::node_from_bytes_backrefs;
-        use crate::Allocator;
-        use rstest::rstest;
```

Comment:
this inner `test` module seemed like a copy-paste error. I removed it, so this is best reviewed ignoring whitespace